### ISSUE TITLE
Spotify Provider OAuth Parse Profile Error: TypeError: Cannot read property 'url' of undefined

### ISSUE THREAD

**Issue by frankmeszaros** [nextauthjs/next-auth#890]
**Describe the bug**
As a spotify user that does not have any profile images, I cannot successfully authenticate with Spotify due to an undefined corner-case error.

**Steps to reproduce**
1. Copy the next-auth example project 
1. [Add Spotify provider](https://next-auth.js.org/providers/spotify) to `api/auth/[...nextauth].js`
1. `cp .env.local.example env.local`, and be sure to supply values for `SPOTIFY_ID` and `SPOTIFY_SECRET`

Repository with bug: https://github.com/frankmeszaros/nextauthspotifybug

**Expected behavior**
Given a s`SPOTIFY_ID` and `SPOTIFY_SECRET` to `.env.local`, I should be able to start the next-auth example project and login with spotify credentials.

**Screenshots or error logs**
![image](https://user-images.githubusercontent.com/15343791/100518489-ce6f8000-315f-11eb-9809-1f04a7c298c7.png)

**Additional context**

I had attempted to pull `next-auth` locally to see if I could submit a small PR to fix this, but when I linked my project to a local copy of `next-auth`, the example application worked with no problem, which was even more confusing. 

After further inspecting, it looks like when I'm linking to the local project this is the snippet that gets hit when I link to `next-auth` locally:
```
profile: (profile) => {
			return {
				id: profile.id,
				name: profile.display_name,
				email: profile.email,
				image: profile.images.length > 0 ? profile.images[0].url : undefined,
			};
		},
```

Because of this, when I link to `next-auth` locally, the project behaves fine. 

Since I got it working locally, i tried double-checking to make sure I wasn't crazy, but the error still persisted when i went back to the version of `next-auth` that comes from NPM. 

When i inspected the stacktrace, it leads me first to `/Users/frankmeszaros/dev/side_projects/songrec/node_modules/next-auth/dist/providers/spotify.js:32:35`, which after further inspection contains the following snippet:

```
profile: _profile => {
      return {
        id: _profile.id,
        name: _profile.display_name,
        email: _profile.email,
        image: _profile.images[0].url
      };
    }
```

__Edit: As a short-term workaround, I added a profile picture to my spotify profile_

**Feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [x] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [ ] Could not find relevant documentation
* [x] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by balazsorban44** [comment]
Hi @frankmeszaros! Thanks for raising the issue. As a workaround, you could change the following:
https://github.com/frankmeszaros/nextauthspotifybug/blob/c06c77e3f9221a6f0b24315dfcefed3d25436c31/pages/api/auth/%5B...nextauth%5D.js#L39-L42

to this:
```js
Providers.Spotify({
  clientId: process.env.SPOTIFY_ID,
  clientSecret: process.env.SPOTIFY_SECRET,
  profile(profile) {
    return {
      id: profile.id,
      name: profile.display_name,
      email: profile.email,
      image: profile.images?.[0]?.url
    }
  },
}),
```
Any user-provided properties added to `Providers.Spotify` will override the default one. This uses [Optional Chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining), which is available out-of-the-box since [Next.js 9.1.7](https://nextjs.org/blog/next-9-1-7#support-for-the-latest-javascript-features).

I am going to apply this fix in `next-auth` through #914. You should be able to remove this, when this is merged and published to npm.

**Comment by frankmeszaros** [comment]
Hey @balazsorban44 thank you for your reply! This must've gotten lost in the ether for me. I was able to work around this using a custom connection as you previously eluded to. Thank you for getting this fix in! Us Spotify users who are too lazy to add a profile picture appreciate it :)
